python/geom.py

```diff
@@ -237,31 +240,37 @@ def transform(self, m):
+    
+    def mu(self,freq):
+        return self._get_epsmu(self.mu_diag, self.mu_offdiag, self.H_susceptibilities, self.B_conductivity_diag, self.B_conductivity_offdiag, freq)
+    
+    def _get_epsmu(self, diag, offdiag, susceptibilities, conductivity_diag, conductivity_offdiag, freq):
         # Clean the input
-        if not isinstance(freq, list):
-            raise ValueError('Frequencies must be supplied as a list')
+        if type(freq) is not np.ndarray:
+            freqs = np.array(freq)
+        else:
+            freqs = np.squeeze(freq)
+
+        numFreqs = freqs.size
 
         # Initialize with instantaneous dielectric tensor
-        eps = [Matrix(mp.Vector3(self.epsilon_diag.x, self.epsilon_offdiag.x, self.epsilon_offdiag.y),
-                     mp.Vector3(self.epsilon_offdiag.x, self.epsilon_diag.y, self.epsilon_offdiag.z),
-                     mp.Vector3(self.epsilon_offdiag.y, self.epsilon_offdiag.z, self.epsilon_diag.z)) for i in range(len(freq))]
-
-        # Iterate through frequencies
-        for i_freq in range(len(freq)):
-            # Iterate through susceptibilities
-            for i_sus in range(len(self.E_susceptibilities)):
-                eps[i_freq] = eps[i_freq] + self.E_susceptibilities[i_sus].eval_susceptibility(freq[i_freq])
-            # Account for conductivity term
-            D_conductivity = Matrix(mp.Vector3(self.D_conductivity_diag.x, self.D_conductivity_offdiag.x, self.D_conductivity_offdiag.y),
-                       mp.Vector3(self.D_conductivity_offdiag.x, self.D_conductivity_diag.y, self.D_conductivity_offdiag.z),
-                       mp.Vector3(self.D_conductivity_offdiag.y, self.D_conductivity_offdiag.z, self.D_conductivity_diag.z))
-            ones_matrix = Matrix(mp.Vector3(1, 1, 1),
-                       mp.Vector3(1, 1, 1),
-                       mp.Vector3(1, 1, 1))
-            eps[i_freq] = (ones_matrix + 1j/freq[i_freq] * D_conductivity) * eps[i_freq]
+        epsmu = np.tile(np.array(Matrix(mp.Vector3(diag.x, offdiag.x, offdiag.y),
+                     mp.Vector3(offdiag.x, diag.y, offdiag.z),
+                     mp.Vector3(offdiag.y, offdiag.z, diag.z))),(numFreqs,1,1))
+
+        # Iterate through susceptibilities
+        for i_sus in range(len(susceptibilities)):
+            epsmu = epsmu + susceptibilities[i_sus].eval_susceptibility(freqs)
         
-        return eps
+        # Account for conductivity term
+        conductivity = np.tile(np.array(Matrix(mp.Vector3(conductivity_diag.x, conductivity_offdiag.x, conductivity_offdiag.y),
+                    mp.Vector3(conductivity_offdiag.x, conductivity_diag.y, conductivity_offdiag.z),
+                    mp.Vector3(conductivity_offdiag.y, conductivity_offdiag.z, conductivity_diag.z))),(numFreqs,1,1))
+        epsmu = (1 + 1j/np.tile(freqs,(3,3,1)).T * conductivity) * epsmu
```

Comment:
`conductivity` and `epsmu` have size `(1,3,3)`, then `epsmu = (1 + 1j/np.reshape(freqs,(numFreqs,1,1)) * conductivity) * epsmu` should I think have the desired result due to broadcasting (no need for `np.tile`).

Reply:
`conductivity` and `epsmu` are initially size (3,3), which is why I was using tile. But it makes more sense to add singleton dimensions and let `numpy` handle the broadcasting. I'll add this in the next commit as well.